source/module_hsolver/diago_dav_subspace.cpp

```diff
@@ -445,6 +445,8 @@ void Diago_DavSubspace<T, Device>::cal_elem(const int& dim,
     if (this->diag_comm.nproc > 1)
     {
         auto* swap = new T[notconv * this->nbase_x];
+        auto* target = new T[notconv * this->nbase_x];
```

Comment:
update again to not affecting the original code?

Reply:
Updated this in the latest commit and added some code. It is ok to merge after getting the tests passed.